dropshot/src/handler.rs

```diff
@@ -1141,13 +1141,20 @@ where
         ApiEndpointResponse {
             schema: Some(ApiSchemaGenerator::Gen {
                 name: T::Body::schema_name,
-                schema: T::Body::json_schema,
+                schema: make_subschema_for::<T::Body>,
             }),
             success: Some(T::STATUS_CODE),
             description: Some(T::DESCRIPTION.to_string()),
         }
     }
 }
+
+fn make_subschema_for<T: JsonSchema>(
+    gen: &mut schemars::gen::SchemaGenerator,
+) -> schemars::schema::Schema {
+    gen.subschema_for::<T>()
```

Comment:
nice find; I hadn't seen that method previously

Reply:
Heh, I only found it when digging into why non-top-level types _did_ create references instead of inline schemas.